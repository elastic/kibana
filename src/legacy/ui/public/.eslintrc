rules:
  no-console: 2
  'import/no-default-export': error
