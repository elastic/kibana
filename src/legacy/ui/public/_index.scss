// Legacy styles to come before all
@import './styles/_legacy/index';

// Prefix all styles with "kbn" to avoid conflicts.
// Examples
//   kbnChart
//   kbnChart__legend
//   kbnChart__legend--small
//   kbnChart__legend-isLoading

@import './accessibility/index';
@import './directives/index';
@import './error_url_overflow/index';
@import './field_editor/index';
@import '../../../plugins/management/public/components/index';
