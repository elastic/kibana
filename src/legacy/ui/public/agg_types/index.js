/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import './agg_params';
import { IndexedArray } from '../indexed_array';
import { countMetricAgg } from './metrics/count';
import { avgMetricAgg } from './metrics/avg';
import { sumMetricAgg } from './metrics/sum';
import { medianMetricAgg } from './metrics/median';
import { minMetricAgg } from './metrics/min';
import { maxMetricAgg } from './metrics/max';
import { topHitMetricAgg } from './metrics/top_hit';
import { stdDeviationMetricAgg } from './metrics/std_deviation';
import { cardinalityMetricAgg } from './metrics/cardinality';
import { percentilesMetricAgg } from './metrics/percentiles';
import { geoBoundsMetricAgg } from './metrics/geo_bounds';
import { geoCentroidMetricAgg } from './metrics/geo_centroid';
import { percentileRanksMetricAgg } from './metrics/percentile_ranks';
import { derivativeMetricAgg } from './metrics/derivative';
import { cumulativeSumMetricAgg } from './metrics/cumulative_sum';
import { movingAvgMetricAgg } from './metrics/moving_avg';
import { serialDiffMetricAgg } from './metrics/serial_diff';
import { dateHistogramBucketAgg } from './buckets/date_histogram';
import { histogramBucketAgg } from './buckets/histogram';
import { rangeBucketAgg } from './buckets/range';
import { dateRangeBucketAgg } from './buckets/date_range';
import { ipRangeBucketAgg } from './buckets/ip_range';
import { termsBucketAgg } from './buckets/terms';
import { filterBucketAgg } from './buckets/filter';
import { filtersBucketAgg } from './buckets/filters';
import { significantTermsBucketAgg } from './buckets/significant_terms';
import { geoHashBucketAgg } from './buckets/geo_hash';
import { geoTileBucketAgg } from './buckets/geo_tile';
import { bucketSumMetricAgg } from './metrics/bucket_sum';
import { bucketAvgMetricAgg } from './metrics/bucket_avg';
import { bucketMinMetricAgg } from './metrics/bucket_min';
import { bucketMaxMetricAgg } from './metrics/bucket_max';

const aggs = {
  metrics: [
    countMetricAgg,
    avgMetricAgg,
    sumMetricAgg,
    medianMetricAgg,
    minMetricAgg,
    maxMetricAgg,
    stdDeviationMetricAgg,
    cardinalityMetricAgg,
    percentilesMetricAgg,
    percentileRanksMetricAgg,
    topHitMetricAgg,
    derivativeMetricAgg,
    cumulativeSumMetricAgg,
    movingAvgMetricAgg,
    serialDiffMetricAgg,
    bucketAvgMetricAgg,
    bucketSumMetricAgg,
    bucketMinMetricAgg,
    bucketMaxMetricAgg,
    geoBoundsMetricAgg,
    geoCentroidMetricAgg
  ],
  buckets: [
    dateHistogramBucketAgg,
    histogramBucketAgg,
    rangeBucketAgg,
    dateRangeBucketAgg,
    ipRangeBucketAgg,
    termsBucketAgg,
    filterBucketAgg,
    filtersBucketAgg,
    significantTermsBucketAgg,
    geoHashBucketAgg,
    geoTileBucketAgg,
  ]
};

Object.keys(aggs).forEach(function (type) {
  aggs[type].forEach(function (agg) {
    agg.type = type;
  });
});


/**
 * IndexedArray of Aggregation Types.
 *
 * These types form two groups, metric and buckets.
 *
 * @module agg_types
 * @type {IndexedArray}
 */
export const aggTypes = new IndexedArray({

  /**
   * @type {Array}
   */
  index: ['name'],

  /**
   * [group description]
   * @type {Array}
   */
  group: ['type'],
  initialSet: aggs.metrics.concat(aggs.buckets)
});
