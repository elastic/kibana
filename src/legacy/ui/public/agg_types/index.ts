/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

// @ts-ignore
import { AggType } from './agg_type';
// @ts-ignore
import { countMetricAgg } from './metrics/count';
// @ts-ignore
import { avgMetricAgg } from './metrics/avg';
// @ts-ignore
import { sumMetricAgg } from './metrics/sum';
// @ts-ignore
import { medianMetricAgg } from './metrics/median';
// @ts-ignore
import { minMetricAgg } from './metrics/min';
// @ts-ignore
import { maxMetricAgg } from './metrics/max';
// @ts-ignore
import { topHitMetricAgg } from './metrics/top_hit';
// @ts-ignore
import { stdDeviationMetricAgg } from './metrics/std_deviation';
// @ts-ignore
import { cardinalityMetricAgg } from './metrics/cardinality';
// @ts-ignore
import { percentilesMetricAgg } from './metrics/percentiles';
// @ts-ignore
import { geoBoundsMetricAgg } from './metrics/geo_bounds';
// @ts-ignore
import { geoCentroidMetricAgg } from './metrics/geo_centroid';
// @ts-ignore
import { percentileRanksMetricAgg } from './metrics/percentile_ranks';
// @ts-ignore
import { derivativeMetricAgg } from './metrics/derivative';
// @ts-ignore
import { cumulativeSumMetricAgg } from './metrics/cumulative_sum';
// @ts-ignore
import { movingAvgMetricAgg } from './metrics/moving_avg';
// @ts-ignore
import { serialDiffMetricAgg } from './metrics/serial_diff';
// @ts-ignore
import { dateHistogramBucketAgg, setBounds } from './buckets/date_histogram';
// @ts-ignore
import { histogramBucketAgg } from './buckets/histogram';
// @ts-ignore
import { rangeBucketAgg } from './buckets/range';
// @ts-ignore
import { dateRangeBucketAgg } from './buckets/date_range';
// @ts-ignore
import { ipRangeBucketAgg } from './buckets/ip_range';
// @ts-ignore
import { termsBucketAgg } from './buckets/terms';
// @ts-ignore
import { filterBucketAgg } from './buckets/filter';
// @ts-ignore
import { filtersBucketAgg } from './buckets/filters';
// @ts-ignore
import { significantTermsBucketAgg } from './buckets/significant_terms';
// @ts-ignore
import { geoHashBucketAgg } from './buckets/geo_hash';
// @ts-ignore
import { geoTileBucketAgg } from './buckets/geo_tile';
// @ts-ignore
import { bucketSumMetricAgg } from './metrics/bucket_sum';
// @ts-ignore
import { bucketAvgMetricAgg } from './metrics/bucket_avg';
// @ts-ignore
import { bucketMinMetricAgg } from './metrics/bucket_min';
// @ts-ignore
import { bucketMaxMetricAgg } from './metrics/bucket_max';

export const aggTypes = {
  metrics: [
    countMetricAgg,
    avgMetricAgg,
    sumMetricAgg,
    medianMetricAgg,
    minMetricAgg,
    maxMetricAgg,
    stdDeviationMetricAgg,
    cardinalityMetricAgg,
    percentilesMetricAgg,
    percentileRanksMetricAgg,
    topHitMetricAgg,
    derivativeMetricAgg,
    cumulativeSumMetricAgg,
    movingAvgMetricAgg,
    serialDiffMetricAgg,
    bucketAvgMetricAgg,
    bucketSumMetricAgg,
    bucketMinMetricAgg,
    bucketMaxMetricAgg,
    geoBoundsMetricAgg,
    geoCentroidMetricAgg,
  ],
  buckets: [
    dateHistogramBucketAgg,
    histogramBucketAgg,
    rangeBucketAgg,
    dateRangeBucketAgg,
    ipRangeBucketAgg,
    termsBucketAgg,
    filterBucketAgg,
    filtersBucketAgg,
    significantTermsBucketAgg,
    geoHashBucketAgg,
    geoTileBucketAgg,
  ],
};

aggTypes.metrics.forEach(aggType => (aggType.type = 'metrics'));
aggTypes.buckets.forEach(aggType => (aggType.type = 'buckets'));

export { AggParam } from './agg_params';
export { AggType } from './agg_type';
export { FieldParamType } from './param_types';
export { BucketAggType } from './buckets/_bucket_agg_type';
export { setBounds };
