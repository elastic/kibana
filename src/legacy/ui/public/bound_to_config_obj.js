/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import _ from 'lodash';

export function BoundToConfigObjProvider(config) {
  /**
   * Create an object with properties that may be bound to config values.
   * The input object is basically cloned unless one of it's own properties
   * resolved to a string value that starts with an equal sign. When that is
   * found, that property is forever bound to the corresponding config key.
   *
   * example:
   *
   * // name is cloned, height is bound to the defaultHeight config key
   * { name: 'john', height: '=defaultHeight' };
   *
   * @param  {Object} input
   * @return {Object}
   */
  function BoundToConfigObj(input) {
    const self = this;

    _.forOwn(input, function(value, prop) {
      if (!_.isString(value) || value.charAt(0) !== '=') {
        self[prop] = value;
        return;
      }

      const configKey = value.substr(1);

      config.watch(configKey, function update(value) {
        self[prop] = value;
      });
    });
  }

  return BoundToConfigObj;
}
