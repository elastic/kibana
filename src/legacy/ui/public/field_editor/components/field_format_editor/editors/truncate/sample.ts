/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

export const sample = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae sem consequat, sollicitudin enim a, feugiat mi. Curabitur congue laoreet elit, eu dictum nisi commodo ut. Nullam congue sem a blandit commodo. Suspendisse eleifend sodales leo ac hendrerit. Nam fringilla tempor fermentum. Ut tristique pharetra sapien sit amet pharetra. Ut turpis massa, viverra id erat quis, fringilla vehicula risus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Phasellus tincidunt gravida gravida. Praesent et ligula viverra, semper lacus in, tristique elit. Cras ac eleifend diam. Nulla facilisi. Morbi id sagittis magna. Sed fringilla, magna in suscipit aliquet."; // eslint-disable-line
