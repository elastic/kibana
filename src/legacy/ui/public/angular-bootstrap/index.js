
/* eslint-disable */

/**
 * TODO: Write custom components that address our needs to directly and deprecate these Bootstrap components.
 */

import 'angular';

import { uiModules } from 'ui/modules';

uiModules.get('kibana', [
  'ui.bootstrap',
]);

/*
 * angular-ui-bootstrap
 * http://angular-ui.github.io/bootstrap/

 * Version: 0.12.1 - 2015-02-20
 * License: MIT
 */
angular.module('ui.bootstrap', [
  'ui.bootstrap.tpls',
  'ui.bootstrap.transition',
  'ui.bootstrap.collapse',
  'ui.bootstrap.alert',
  'ui.bootstrap.bindHtml',
  'ui.bootstrap.buttons',
  'ui.bootstrap.dateparser',
  'ui.bootstrap.position',
  'ui.bootstrap.dropdown',
  'ui.bootstrap.modal',
  'ui.bootstrap.pagination',
  'ui.bootstrap.tooltip',
  'ui.bootstrap.popover',
  'ui.bootstrap.progressbar',
  'ui.bootstrap.rating',
  'ui.bootstrap.tabs',
  'ui.bootstrap.timepicker',
  'ui.bootstrap.typeahead'
]);

angular.module('ui.bootstrap.tpls', [
  'template/alert/alert.html',
  'template/modal/backdrop.html',
  'template/modal/window.html',
  'template/pagination/pager.html',
  'template/pagination/pagination.html',
  'template/tooltip/tooltip-html-unsafe-popup.html',
  'template/tooltip/tooltip-popup.html',
  'template/popover/popover.html',
  'template/progressbar/bar.html',
  'template/progressbar/progress.html',
  'template/progressbar/progressbar.html',
  'template/rating/rating.html',
  'template/tabs/tab.html',
  'template/tabs/tabset.html',
  'template/timepicker/timepicker.html',
  'template/typeahead/typeahead-match.html',
  'template/typeahead/typeahead-popup.html'
]);

import './accordion/accordion';
import './alert/alert';
import './bindHtml/bindHtml';
import './buttons/buttons';
import './collapse/collapse';
import './dateparser/dateparser';
import './dropdown/dropdown';
import './modal/modal';
import './pagination/pagination';
import './popover/popover';
import './position/position';
import './progressbar/progressbar';
import './rating/rating';
import './tabs/tabs';
import './timepicker/timepicker';
import './tooltip/tooltip';
import './transition/transition';
import './typeahead/typeahead';

import alert from './alert/alert.html';

angular.module('template/alert/alert.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/alert/alert.html', alert);
}]);

import backdrop from './modal/backdrop.html';

angular.module('template/modal/backdrop.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/modal/backdrop.html', backdrop);
}]);

import modal from './modal/window.html';

angular.module('template/modal/window.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/modal/window.html', modal);
}]);

import pager from './pagination/pager.html';

angular.module('template/pagination/pager.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/pagination/pager.html', pager);
}]);

import pagination from './pagination/pagination.html';

angular.module('template/pagination/pagination.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/pagination/pagination.html', pagination);
}]);

import tooltipUnsafePopup from './tooltip/tooltip-html-unsafe-popup.html';

angular.module('template/tooltip/tooltip-html-unsafe-popup.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/tooltip/tooltip-html-unsafe-popup.html', tooltipUnsafePopup);
}]);

import tooltipPopup from './tooltip/tooltip-popup.html';

angular.module('template/tooltip/tooltip-popup.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/tooltip/tooltip-popup.html', tooltipPopup);
}]);

import popover from './popover/popover.html';

angular.module('template/popover/popover.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/popover/popover.html', popover);
}]);

import bar from './progressbar/bar.html';

angular.module('template/progressbar/bar.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/progressbar/bar.html', bar);
}]);

import progress from './progressbar/progress.html';

angular.module('template/progressbar/progress.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/progressbar/progress.html', progress);
}]);

import progressbar from './progressbar/progressbar.html';

angular.module('template/progressbar/progressbar.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/progressbar/progressbar.html', progressbar);
}]);

import rating from './rating/rating.html';

angular.module('template/rating/rating.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/rating/rating.html', rating);
}]);

import tab from './tabs/tab.html';

angular.module('template/tabs/tab.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/tabs/tab.html', tab);
}]);

import tabset from './tabs/tabset.html';

angular.module('template/tabs/tabset.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/tabs/tabset.html', tabset);
}]);

import timepicker from './timepicker/timepicker.html';

angular.module('template/timepicker/timepicker.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/timepicker/timepicker.html', timepicker);
}]);

import typeaheadMatch from './typeahead/typeahead-match.html';

angular.module('template/typeahead/typeahead-match.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/typeahead/typeahead-match.html', typeaheadMatch);
}]);

import typeaheadPopup from './typeahead/typeahead-popup.html';

angular.module('template/typeahead/typeahead-popup.html', []).run(['$templateCache', function($templateCache) {
  $templateCache.put('template/typeahead/typeahead-popup.html', typeaheadPopup);
}]);
