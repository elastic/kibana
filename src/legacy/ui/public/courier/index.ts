/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

/**
 * Nothing to see here!
 *
 * Courier / SearchSource has moved to the data plugin, and is being
 * re-exported from ui/courier for backwards compatibility.
 */

import { start as dataStart } from '../../../core_plugins/data/public/legacy';

// runtime contracts
export const { defaultSearchStrategy, SearchSource } = dataStart.search;

// types
export {
  ISearchSource,
  EsQuerySortValue, // used externally by Discover
  FetchOptions, // used externally by AggTypes
  SortDirection, // used externally by Discover
} from '../../../core_plugins/data/public';

// static code
export {
  getRequestInspectorStats,
  getResponseInspectorStats,
} from '../../../core_plugins/data/public';

// TODO: Exporting this mock outside of jest tests causes errors because
// jest is undefined. Need to refactor the mock to be consistent with
// other NP-style mocks.
// export { searchSourceMock } from './search_source/mocks';

// Most these can probably be made internal to the search
// service, so we are temporarily deeply importing them
// until we relocate them to a longer-term home.
/* eslint-disable @kbn/eslint/no-restricted-paths */
export {
  addSearchStrategy, // used externally by Rollups
  getSearchErrorType, // used externally by Rollups
  hasSearchStategyForIndexPattern, // used externally by Discover
  isDefaultTypeIndexPattern, // used externally by Discover
  SearchError, // used externally by Visualizations & Rollups
} from '../../../core_plugins/data/public/search/search_strategy';
/* eslint-enable @kbn/eslint/no-restricted-paths */
