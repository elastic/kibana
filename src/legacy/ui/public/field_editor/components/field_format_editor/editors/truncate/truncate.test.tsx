/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import React, { ChangeEvent } from 'react';
import { shallow } from 'enzyme';
import { EuiFieldNumber } from '@elastic/eui';
import { FieldFormat } from 'src/plugins/data/public';

import { TruncateFormatEditor } from './truncate';

const fieldType = 'string';
const format = {
  getConverterFor: jest.fn().mockImplementation(() => (input: string) => input.substring(0, 10)),
  getParamDefaults: jest.fn().mockImplementation(() => {
    return { fieldLength: 10 };
  }),
};
const formatParams = {
  fieldLength: 5,
};
const onChange = jest.fn();
const onError = jest.fn();

describe('TruncateFormatEditor', () => {
  beforeEach(() => {
    onChange.mockClear();
    onError.mockClear();
  });

  it('should have a formatId', () => {
    expect(TruncateFormatEditor.formatId).toEqual('truncate');
  });

  it('should render normally', async () => {
    const component = shallow(
      <TruncateFormatEditor
        basePath={''}
        fieldType={fieldType}
        format={(format as unknown) as FieldFormat}
        formatParams={formatParams}
        onChange={onChange}
        onError={onError}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fire error, when input is invalid', async () => {
    const component = shallow(
      <TruncateFormatEditor
        basePath={''}
        fieldType={fieldType}
        format={(format as unknown) as FieldFormat}
        formatParams={formatParams}
        onChange={onChange}
        onError={onError}
      />
    );
    const input = component.find(EuiFieldNumber);

    const changeEvent = {
      target: {
        value: '123.3',
        checkValidity: () => false,
        validationMessage: 'Error!',
      },
    };

    await input!.invoke('onChange')!((changeEvent as unknown) as ChangeEvent<HTMLInputElement>);

    expect(onError).toBeCalledWith(changeEvent.target.validationMessage);
    expect(onChange).not.toBeCalled();
  });

  it('should fire change, when input changed and is valid', async () => {
    const component = shallow(
      <TruncateFormatEditor
        basePath={''}
        fieldType={fieldType}
        format={(format as unknown) as FieldFormat}
        formatParams={formatParams}
        onChange={onChange}
        onError={onError}
      />
    );
    const input = component.find(EuiFieldNumber);

    const changeEvent = {
      target: {
        value: '123',
        checkValidity: () => true,
        validationMessage: null,
      },
    };
    onError.mockClear();
    await input!.invoke('onChange')!((changeEvent as unknown) as ChangeEvent<HTMLInputElement>);
    expect(onError).not.toBeCalled();
    expect(onChange).toBeCalledWith({ fieldLength: 123 });
  });
});
