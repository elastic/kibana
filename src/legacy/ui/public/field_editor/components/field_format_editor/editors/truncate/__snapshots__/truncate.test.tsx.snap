// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruncateFormatEditor should render normally 1`] = `
<Fragment>
  <EuiFormRow
    describedByIds={Array []}
    display="row"
    fullWidth={false}
    hasChildLabel={true}
    hasEmptyLabelSpace={false}
    isInvalid={false}
    label={
      <FormattedMessage
        defaultMessage="Field length"
        id="common.ui.fieldEditor.truncate.lengthLabel"
        values={Object {}}
      />
    }
    labelType="label"
  >
    <EuiFieldNumber
      defaultValue={5}
      isInvalid={false}
      min={1}
      onChange={[Function]}
    />
  </EuiFormRow>
  <FormatEditorSamples
    sampleType="text"
    samples={
      Array [
        Object {
          "input": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris vitae sem consequat, sollicitudin enim a, feugiat mi. Curabitur congue laoreet elit, eu dictum nisi commodo ut. Nullam congue sem a blandit commodo. Suspendisse eleifend sodales leo ac hendrerit. Nam fringilla tempor fermentum. Ut tristique pharetra sapien sit amet pharetra. Ut turpis massa, viverra id erat quis, fringilla vehicula risus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Phasellus tincidunt gravida gravida. Praesent et ligula viverra, semper lacus in, tristique elit. Cras ac eleifend diam. Nulla facilisi. Morbi id sagittis magna. Sed fringilla, magna in suscipit aliquet.",
          "output": "Lorem ipsu",
        },
      ]
    }
  />
</Fragment>
`;
